src/form.js

```diff
@@ -187,3 +187,24 @@ export function isFieldDefault(field) {
   }
   return true;
 }
+
+/**
+ * Checks if a form field is empty.
+ * @param {!Element} field
+ * @return {boolean}
+ */
+export function isFieldEmpty(field) {
+  switch (field.type) {
+    case 'select-multiple':
+    case 'select-one':
+      // dropdown menu has at least one option always selected
+      return false;
+    case 'checkbox':
+    case 'radio':
+      return !field.checked;
+    default:
```

Comment:
Let's do an exhaustive switch and throw an error by default. (and also a test for the default case)

Reply:
Do you think an enumeration of all `type`s is necessary? Or could we get away with checking if it's an `input` element? (There are ~20 `type`s valid in AMP right now)

Reply:
Okay now I understand why they don't have an exhaustive switch in the other helper......

However, as you point out, I am still confused why we don't check elements first. Both this and `isFieldDefault` helpers don't make sense on elements other than <input> <select> and <amp-selector> and etc., but the call site didn't specify.

I suggest we switch on the element tag names:

```js
switch(element.tagName):
  case input:
     if filed.type === radio or field.type === radio
        return ...;
     else
        return ...;
  case select:
     return false;
  case amp-selector:
    //TODO
  default:
     return false;
```

But I might be missing something, in that case, we should at least document in the JsDoc what kind of elements which be called with this helper.

Reply:
Not sure if it's a bug or never supported, I can't put code snippets with Md here. Anyway, code snippets:
https://jsfiddle.net/698aqzLx/